### PR TITLE
Rename scalar/basic_scipy to scalar/math

### PR DESCRIPTION
**Pull request by fanshi118**
**Thank you for opening a PR!**

Here are a few important guidelines and requirements to check before your PR can be merged:
+ [ ] There is an informative high-level description of the changes.
+ [ ] The description and/or commit message(s) references the relevant GitHub issue(s).
+ [ ] [`pre-commit`](https://pre-commit.com/#installation) is installed and [set up](https://pre-commit.com/#3-install-the-git-hook-scripts).
+ [ ] The commit messages follow [these guidelines](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html).
+ [ ] The commits correspond to [_relevant logical changes_](https://wiki.openstack.org/wiki/GitCommitMessages#Structural_split_of_changes), and there are **no commits that fix changes introduced by other commits in the same branch/BR**.  If your commit description starts with "Fix...", then you're probably making this mistake.
+ [ ] There are tests covering the changes introduced in the PR.

Don't worry, your PR doesn't need to be in perfect order to submit it.  As development progresses and/or reviewers request changes, you can always [rewrite the history](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History#_rewriting_history) of your feature/PR branches.

If your PR is an ongoing effort and you would like to involve us in the process, simply make it a [draft PR](https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/about-pull-requests#draft-pull-requests).

### CODE REVIEW

**Comment by ricardoV94** [comment]
@fanshi, thanks for opening the PR. It seems that there are quite more places where the imports need to be changed, gauging from the failing tests. You will also have to check whether any doc examples make reference to the affected imports.

**Comment by fanshi118** [comment]
> @fanshi, thanks for opening the PR. It seems that there are more places where the imports need to be changed, gauging from the failing tests. You will also have to check whether any doc examples make reference to the affected imports.

Thanks for reviewing. I made some additional changes in places I found needed. Let me know how this looks!

File: tests/link/test_numba.py [hunk 1]
```diff
@@ -8,7 +8,7 @@
 
 import aesara.scalar as aes
 import aesara.scalar.basic as aesb
-import aesara.scalar.basic_scipy as aes_sci
+import aesara.scalar.math as aes_sci
```
Comment (ricardoV94) [inline tests/link/test_numba.py:11]: This looks like it should be `import xxx as aesm` by looking at the others

Comment (fanshi118) [reply tests/link/test_numba.py:11]: Sounds good. Will update.

**Comment by ricardoV94** [comment]
@fanshi118 I made two extra fixes in `tests/tensor/test_math_opt.py` where there was still a `aes.basic_scipy.imported_scipy_special`. I also squashed your commits into a single one, so if you need to do any further changes you'll have to rebase from here

**Review by ricardoV94 — APPROVED** [review]
LGTM, hopefully we didn't miss anything :) 

Thanks for opening the PR

**Review by ricardoV94 — APPROVED** [review]

**Comment by fanshi118** [comment]
> @fanshi118 I made two extra fixes in `tests/tensor/test_math_opt.py` where there was still a `aes.basic_scipy.imported_scipy_special`. I also squashed your commits into a single one, so if you need to do any further changes you'll have to rebase from here.
> 
> Edit: And just rebased from master

Ah I missed those. Thanks for coming through!

**Comment by ricardoV94** [comment]
Thanks @fanshi118!

**Comment by brandonwillard** [comment]
This branch appears to have had two merge commits.  We don't want those kinds of commits in the main branch.

**Comment by ricardoV94** [comment]
I rebased when merging assuming it would "erase" those.

**Comment by brandonwillard** [comment]
It did; I should have prefaced that with "warning".

**Pull request merged**